public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx

```diff
@@ -0,0 +1,257 @@
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  ViewChild,
+  ViewEncapsulation
+} from '@angular/core';
+import { environment } from 'environments/environment';
+import * as highCharts from 'highcharts';
+import HighchartsReact from 'highcharts-react-official';
+import * as React from 'react';
+import { FunctionComponent } from 'react';
+import * as ReactDOM from 'react-dom';
+import { BaseReactWrapper } from '../../../data-browser/base-react/base-react.wrapper';
+import { baseOptions, GENDER_STRATUM_MAP } from '../../../data-browser/charts/react-base-chart/base-chart.service';
+
+const containerElementName = 'root';
+
+interface State {
+    options: any;
+}
+
+interface Props {
+    genderAnalysis: any;
+    genderCountAnalysis: any;
+    selectedResult: any;
+    domain: string;
+}
+
+export class BioSexChartReactComponent extends React.Component<Props, State> {
+
+  constructor(props) {
+    super(props);
+    this.state = {options: null};
+  }
+
+  componentDidMount() {
+      this.getChartOptions();
+  }
+
+  getChartOptions() {
+    if (this.props.domain === 'fitbit') {
+        this.getFitbitChartOptions();
+    } else {
+        this.getSurveyChartOptions();
+    }
+  }
+
+  getFitbitChartOptions() {
+    baseOptions.chart.type = 'bar';
+    baseOptions.plotOptions.series.pointWidth = 50;
+    baseOptions.yAxis.title.text = 'Participant Count';
+    baseOptions.xAxis.title.text = '';
+    baseOptions.yAxis.title.style.fontSize = '14px';
+    baseOptions.xAxis.title.style.fontSize = '14px';
+    baseOptions.yAxis.title.style.color = '#262262';
+    baseOptions.yAxis.gridLineColor = '#F0F0F0';
+    const {categories, data} = this.prepFitbitCategoriesAndData();
+    baseOptions.xAxis.categories = categories;
+    baseOptions.series[0].data = data;
+    this.setState({options: baseOptions});
+  }
+
+  getSurveyChartOptions() {
```

Comment:
Nit: Pull all the props here. Can also pull nested properties to simplify further:
```suggestion
  getSurveyChartOptions() {
    const {genderAnalysis: {analysisName, results}, selectedResult} = this.props;
```

Reply:
Changed this.